fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerUniqueIndexTest.java

```diff
@@ -366,4 +371,184 @@ public void resolveUniquenessViolations() {
             context.commit();
         }
     }
+
+    @Test
+    public void testReadableUniquePendingSingle() {
```

Comment:
It would probably be good to add a test that builds an index off an index that has uniqueness violations.
I'm wondering if there could be risk if the destination index made assumptions about the source index being unique, but the only thing I can think of would be if the destination was the same index definition, you could see the indexer detecting that, and just copying the range, but missing the fact that the source had uniqueness violations, resulting in the destination being readable, but with uniqueness violations.

Reply:
About the index from index - it does not make any uniqueness assumption and will detect uniqueness violations (if applicable) the same way as a by-records scan.